frame.go

```diff
@@ -827,7 +827,11 @@ func (f *framer) writeTo(w io.Writer) error {
 }
 
 func (f *framer) readTrace() {
-	f.traceID = f.readUUID().Bytes()
+	if len(f.buf) < 16 {
+		panic(fmt.Errorf("not enough bytes in buffer to read trace uuid require 16 got: %d", len(f.buf)))
+	}
+	copy(f.traceID, f.buf[:16])
```

Comment:
Copy does not allocate memory:
```suggestion
	if len(f.traceID) != 16 {
		f.traceID = make([]byte, 0, 16)
	}
	copy(f.traceID, f.buf[:16])
```

Please double check it is working.

Reply:
Done with `f.traceID = make([]byte, 16)`